src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,117 @@
 import { useState } from 'react';
+import { Movie } from '../../types/Movie';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+interface MovieForm {
+  title: string;
+  imgUrl: string;
+  imdbUrl: string;
+  imdbId: string;
+  description: string;
+}
+
+export const NewMovie = ({
+  onAddMovie,
+}: {
+  onAddMovie: (movie: Movie) => void;
+}) => {
+  const [form, setForm] = useState<Movie>({
+    title: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+    description: '',
+  });
+
+  const [touched, setTouched] = useState<{ [key: string]: boolean }>({});
+
+  const handleBlur = (name: string) => {
+    setTouched(prev => ({ ...prev, [name]: true }));
+  };
+
+  const handleChange = (name: string, value: string) => {
```

Comment:
```suggestion
  const handleChange = (event) => {
```

Reply:
IT DOESN'T WORK THAT WAY(

Reply:
it should work, you can always ask for some help in the chat

but you also can leave your version, just specify types more precisely
```suggestion
  const handleChange = (name: keyof Movie, value: string) => {
```

Reply:
please fix it 
![Uploading image.png…]()